rust/src/linalg/cosine.rs

```diff
@@ -111,14 +113,31 @@ mod x86_64 {
                     xy = _mm256_fmadd_ps(x, y, xy);
                     y_sq = _mm256_fmadd_ps(y, y, y_sq);
                 }
-                1.0 - add_f32_register(xy) / (x_norm * add_f32_register(y_sq).sqrt())
+                // handle remaining elements
+                let mut dotprod = add_f32_register(xy);
+                dotprod += dotprod_scalar(&x_vector[len..], &y_vector[len..]);
+                let mut y_sq_sum = add_f32_register(y_sq);
+                y_sq_sum += sq_scalar(&y_vector[len..]);
+                1.0 - dotprod / (x_norm * y_sq_sum.sqrt())
             }
         }
     }
 }
 
+#[inline]
+fn dotprod_scalar<T: Real + Sum>(x: &[T], y: &[T]) -> T {
```

Comment:
Lets call https://github.com/eto-ai/lance/blob/main/rust/src/linalg/dot.rs ?

Will add SIMD for dot product later.

Reply:
fixed